core-httpclient-impl/src/main/java/com/optimizely/ab/OptimizelyFactory.java

```diff
@@ -189,6 +189,25 @@ public static Optimizely newDefaultInstance(String sdkKey, String fallback) {
         return newDefaultInstance(builder.build(), notificationCenter);
     }
 
+    /**
+     * Returns a new Optimizely instance with authenticated datafile support.
+     *
+     * @param sdkKey   SDK key used to build the ProjectConfigManager.
+     * @param datafileAuthToken  Token for authenticated datafile access.
+     * @param fallback Fallback datafile string used by the ProjectConfigManager to be immediately available.
+     */
+    public static Optimizely newDefaultInstance(String sdkKey, String datafileAuthToken, String fallback) {
```

Comment:
The cascading signatures here don't line up nicely. We should consistently call the next method in the chain.
```java
public static Optimizely newDefaultInstance(String sdkKey) {
    ...
    return newDefaultInstance(sdkKey, null)
}

public static Optimizely newDefaultInstance(String sdkKey, String fallback) {
    ...
    String datafileAuthToken = PropertyUtils.get(HttpProjectConfigManager.CONFIG_SDK_AUTH_TOKEN);
    return newDefaultInstance(sdkKey, fallback, datafileAuthToken)
}

public static Optimizely newDefaultInstance(String sdkKey, String fallback, String datafileAuthToken) {
    ...
    return newDefaultInstance(builder.build(), notificationCenter);
}
```
Unfortunately these are all strings so order needs to be paid attention to. We should also provide a configuration option for `datafileAuthToken` so customers can deploy this via `optimizely.properties` and not have explicitly provide this value to the factory method. See the example above.